app/questionnaire/navigation.py

```diff
@@ -44,20 +44,22 @@ def build_navigation(self, current_group_id, current_group_instance):
 
             skip_group = self._should_skip_group(current_group_instance, group)
             last_block_in_current_group = group['blocks'][-1]
-            is_summary_or_confirmation = last_block_in_current_group and \
+            is_summary_or_confirm_group = last_block_in_current_group and \
                 last_block_in_current_group['type'] in ('Summary', 'Confirmation')
+            can_get_to_summary = is_summary_or_confirm_group and self.routing_path and \
+                Location(group['id'], 0, last_block_in_current_group['id']) in self.routing_path
```

Comment:
Think we need to extract the Location(group['id'], 0, last_block_in_current_group['id']) out like we do above for first_location, it will makes it more consistent and easy to read, it will also make the if statement nicer